Depricated use of std::result_of in Seq

I've got code that used to compile in VS2015, when I've updated the project to VS2017 (v15.5.1) and v141 platform the code fails to compile. Here is the error message

```
c:\program files (x86)\microsoft visual studio\2017\professional\vc\tools\msvc\14.12.25827\include\type_traits(1808): error C2338: result_of<CallableType> is invalid; use result_of<CallableType(zero or more argument types)> instead. (compiling source file Source Files\Utility\MyTest.cpp)
c:\git\project\include\rapidcheck\seq.hpp(62): note: see reference to class template instantiation 'std::result_of<rc::seq::detail::IterateSeq<unsigned int,rc::seq::index::<lambda_d2e59f21f2928ea5637514f3a710dd17>> (__cdecl *)(void)>' being compiled (compiling source file Source Files\Utility\MyTest.cpp)
c:\git\project\include\rapidcheck\seq\create.hpp(178): note: see reference to function template instantiation 'rc::Seq<::size_t> rc::seq::iterate<::size_t,rc::seq::index::<lambda_d2e59f21f2928ea5637514f3a710dd17>>(T &&,Callable &&)' being compiled
        with
        [
            T=::size_t,
            Callable=rc::seq::index::<lambda_d2e59f21f2928ea5637514f3a710dd17>
        ] (compiling source file Source Files\Utility\MyTest.cpp)
```

The code in question is function
```
template <typename Impl, typename... Args>
Seq<typename std::result_of<Impl()>::type::ValueType> makeSeq(Args &&... args) {
  using SeqT = Seq<typename std::result_of<Impl()>::type::ValueType>;
  using ImplT = typename SeqT::template SeqImpl<Impl>;

  SeqT seq;
  seq.m_impl.reset(new ImplT(std::forward<Args>(args)...));
  return seq;
}
```

Reading the notes of [`result_of`](http://en.cppreference.com/w/cpp/types/result_of), it looks like the current usage has been deprecated with c++17 as it may result in undefined behavior. 

I have quite a bit of code and unsure about a minimal test to demo the issue. I suspect it's something because I use generators, e.g.

```
 template <>
    struct Arbitrary<LARGE_INTEGER>
    {
        static Gen<LARGE_INTEGER> arbitrary()
        {
            return gen::build<LARGE_INTEGER>(
                gen::set(&LARGE_INTEGER::QuadPart));
        }
    };
```

Any thoughts on how I can fix this? Thank you.

--- a/include/rapidcheck/Seq.hpp
+++ b/include/rapidcheck/Seq.hpp
@@ -59,8 +59,8 @@ Seq<T> &Seq<T>::operator=(const Seq &rhs) {
 }
 
 template <typename Impl, typename... Args>
-Seq<typename std::result_of<Impl()>::type::ValueType> makeSeq(Args &&... args) {
-  using SeqT = Seq<typename std::result_of<Impl()>::type::ValueType>;
+Seq<typename std::result_of<Impl(void)>::type::ValueType> makeSeq(Args &&... args) {
+  using SeqT = Seq<typename std::result_of<Impl(void)>::type::ValueType>;
   using ImplT = typename SeqT::template SeqImpl<Impl>;
 
   SeqT seq;

